karapace/schema_reader.py

```diff
@@ -361,7 +361,9 @@ def _delete_schema_below_version(schema, version):
             schema["deleted"] = True
         return schema
 
-    def get_schemas(self, subject):
+    def get_schemas(self, subject, *, include_deleted=False):
```

Comment:
Or just change only this line 

```python
if include_deleted or val.get("deleted", False) is False
```

There'd be only a single execution path.

Reply:
Do you think it would be beneficial as current implementation avoids copying of that list iff `include_deleted=true`?

Reply:
please add types